src/index.js

```diff
@@ -102,39 +132,47 @@ class Dropzone extends React.Component {
 
   onDrop(e) {
     const { onDrop, onDropAccepted, onDropRejected, multiple, disablePreview } = this.props;
-    const fileList = getDataTransferItems(e, multiple);
     const acceptedFiles = [];
     const rejectedFiles = [];
 
     // Stop default browser behavior
     e.preventDefault();
 
-    // Reset the counter along with the drag on a drop.
-    this.enterCounter = 0;
+    // Reset the target list along with the drag on a drop.
+    this.dragTargets = [];
     this.isFileDialogActive = false;
 
-    fileList.forEach((file) => {
-      if (!disablePreview) {
-        file.preview = window.URL.createObjectURL(file); // eslint-disable-line no-param-reassign
-      }
+    // Don't fire callbacks if a child Dropzone component has already handled this event
+    // eslint-disable-next-line no-underscore-dangle
+    if (!e.__dropzoneChildHandledThis) {
```

Comment:
Yeah it's nested Dropzones. An example would be: an instant messenger UI where dragging over an entire messaging window will show an overlay modal, and then you may potentially be able to drop into one or more large icons to indicate whether to send the attachment immediately or compose a message to go with it, with a default behavior of "compose" if the user drops into the modal itself and misses the icons.